File: dotnet_backend/backend/Models/Enrollment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ComputerSeekho.Models;

/// <summary>
/// Places a student in a batch. Carries inquiry_id as well, so the trace
/// back to the original lead survives even if you start from the enrolment.
/// </summary>
[Table("enrollments")]
public class Enrollment
{
    [Key]
    [Column("enrollment_id")]
    public int EnrollmentId { get; set; }

    [Column("student_id")]
    public int StudentId { get; set; }

    [Column("batch_id")]
    public int BatchId { get; set; }

    [Column("inquiry_id")]
    public int InquiryId { get; set; }

    [Column("enroll_date")]
    public DateOnly EnrollDate { get; set; } = DateOnly.FromDateTime(DateTime.Today);

    [Column("status")]
    public EnrollmentStatus Status { get; set; } = EnrollmentStatus.Active;

    [ForeignKey(nameof(StudentId))]
    public Student? Student { get; set; }

    [ForeignKey(nameof(BatchId))]
    public Batch? Batch { get; set; }
}

File: backend/src/main/java/com/smvita/computerseekho/entity/Enrollment.java
package com.smvita.computerseekho.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

/**
 * A student's place in a batch. Carries inquiry_id as well as student_id
 * because the Knowledge Base made "no enquiry, no registration" a hard DB
 * rule (§4.1) — and the BRD's Course-A-then-Course-B example means a repeat
 * student's second enrolment must trace to its own enquiry, not to the one
 * that first brought them in.
 */
@Entity
@Table(name = "enrollments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Enrollment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "enrollment_id")
    private Integer enrollmentId;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "student_id", nullable = false)
    private Student student;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "batch_id", nullable = false)
    private Batch batch;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "inquiry_id", nullable = false)
    private Inquiry inquiry;

    @Column(name = "enroll_date", nullable = false)
    private LocalDate enrollDate = LocalDate.now();

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false,
            columnDefinition = "ENUM('Active','Completed','Dropped')")
    private Status status = Status.Active;

    public enum Status { Active, Completed, Dropped }
}
